### ISSUE TITLE
Api::namespaced should work with default namespace

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#209]
Currently the kube::Config picks up on the default namespace in two ways:
- in cluster: `SERVICE_DEFAULT_NS` evar provided
- local dev: context.namespace.unwrap_or("default")

Which can be passed onto the `kube::Client` so that `Api::namsepaced` and `Resource::namespaced` can omit this if they want a controller to run in the namespace they were deployed to.

**Comment by clux** [comment]
This interplays with the planned api changes in #194.
But we could as a tempoary measure add:

- `Api::default_namespaced()`

Slightly harder to do `Resource::default_namespaced` because the that information is on the `kube::Client`, but that's fine. `Resource` is a lower level API.

**Comment by clux** [comment]
Actually, only doing one of them is awkward for the `Informer` init..

```rust
    let namespace = env::var("NAMESPACE").unwrap_or("default".into());
    let resource = Resource::namespaced::<Pod>(&namespace);
    let inf = Informer::new(client, ListParams::default(), resource);
```

We should probably move Informers and Reflectors to use `Api` rather than `Resource`, and not give it a `Client`. They don't need the lower-level API anyway.

**Comment by clux** [comment]
Stashed diff:

```diff
diff --git kube/src/api/typed.rs kube/src/api/typed.rs
index 16d985e5..996479ea 100644
--- kube/src/api/typed.rs
+++ kube/src/api/typed.rs
@@ -53,6 +53,16 @@ where
             phantom: PhantomData,
         }
     }
+
+    /// Namespaced resource within the configured default namespace
+    pub fn default_namespaced(client: Client) -> Self {
+        let api = Resource::namespaced::<K>(&client.default_ns);
+        Self {
+            api,
+            client,
+            phantom: PhantomData,
+        }
+    }
 }
 
 /// PUSH/PUT/POST/GET abstractions
diff --git kube/src/client/mod.rs kube/src/client/mod.rs
index 8738c0f5..280a535f 100644
--- kube/src/client/mod.rs
+++ kube/src/client/mod.rs
@@ -65,7 +65,7 @@ pub struct Status {
 #[derive(Clone)]
 pub struct Client {
     cluster_url: reqwest::Url,
-    default_ns: String,
+    pub(crate) default_ns: String,
     inner: reqwest::Client,
 }
 
```

**Comment by clux** [comment]
released in 0.56
